code/modules/mob/living/basic/space_fauna/hivebot/_hivebot.dm

```diff
@@ -0,0 +1,138 @@
+	icon = 'icons/mob/simple/hivebot.dmi'
+	icon_state = "basic"
+	icon_living = "basic"
+	icon_dead = "basic"
+	basic_mob_flags = DEL_ON_DEATH
+	gender = NEUTER
+	mob_biotypes = MOB_ROBOTIC
+
+	health = 15
+	maxHealth = 15
+	melee_damage_lower = 2
+	melee_damage_upper = 3
+
+	attack_verb_continuous = "claws"
+	attack_verb_simple = "claw"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	attack_vis_effect = ATTACK_EFFECT_CLAW
+	verb_say = "states"
+	verb_ask = "queries"
+	verb_exclaim = "declares"
+	verb_yell = "alarms"
+	bubble_icon = "machine"
+
+	faction = list(FACTION_HIVEBOT)
+	combat_mode = TRUE
+	speech_span = SPAN_ROBOT
+	death_message = "blows apart!"
+
+	habitable_atmos = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minimum_survivable_temperature = TCMB
+	ai_controller = /datum/ai_controller/basic_controller/hivebot
+	///does this type do range attacks?
+	var/ranged_attacker = FALSE
+
+
+/mob/living/basic/hivebot/Initialize(mapload)
+	. = ..()
+	var/static/list/death_loot = list(/obj/effect/decal/cleanable/robot_debris)
+	AddElement(/datum/element/death_drops, death_loot)
+	AddElement(/datum/element/appearance_on_aggro, overlay_icon = icon, overlay_state = "[initial(icon_state)]_attack")
+	if(!ranged_attacker)
+		return
+	AddElement(/datum/element/ranged_attacks, projectiletype = /obj/projectile/hivebotbullet)
+
+/mob/living/basic/hivebot/death(gibbed)
+	do_sparks(number = 3, cardinal_only = TRUE, source = src)
+	return ..()
+
+/mob/living/basic/hivebot/range
+	name = "hivebot"
+	desc = "A smallish robot, this one is armed!"
+	icon_state = "ranged"
+	icon_living = "ranged"
+	icon_dead = "ranged"
+	ranged_attacker = TRUE
+	ai_controller = /datum/ai_controller/basic_controller/hivebot/ranged
+
+/mob/living/basic/hivebot/rapid
+	icon_state = "ranged"
+	icon_living = "ranged"
+	icon_dead = "ranged"
+	ranged_attacker = TRUE
+	ai_controller = /datum/ai_controller/basic_controller/hivebot/ranged/rapid
+
+/mob/living/basic/hivebot/strong
+	name = "strong hivebot"
+	icon_state = "strong"
+	icon_living = "strong"
+	icon_dead = "strong"
+	desc = "A robot, this one is armed and looks tough!"
+	health = 80
+	maxHealth = 80
+	ranged_attacker = TRUE
+	ai_controller = /datum/ai_controller/basic_controller/hivebot/ranged
+
+/mob/living/basic/hivebot/mechanic
+	name = "hivebot mechanic"
+	icon_state = "strong"
+	icon_living = "strong"
+	icon_dead = "strong"
+	desc = "A robot built for base upkeep, intended for use inside hivebot colonies."
+	health = 60
+	maxHealth = 60
+	gold_core_spawnable = HOSTILE_SPAWN
+	ranged_attacker = TRUE
+	ai_controller = /datum/ai_controller/basic_controller/hivebot/mechanic
+	///the ability to create foam wall
+	var/datum/action/cooldown/spell/conjure/foam_wall/foam
+	///cooldown to repair machines
+	COOLDOWN_DECLARE(repair_cooldown)
+
+/mob/living/basic/hivebot/mechanic/Initialize(mapload)
+	. = ..()
+	foam = new
+	foam.Grant(src)
```

Comment:
```suggestion
	var/datum/action/cooldown/spell/conjure/foam_wall/foam/foam = new(src)
	foam.Grant(src)
```
If you don't need to reference the datum we don't need to save it in a var, creating an action with its target = whatever will automatically clean it up if we're deleted